### DISCUSSION TITLE
Error for MAP++

### DISCUSSION THREAD

**Discussion by sjihf12** [OpenFAST/openfast#1757] [Q&A]
Hi, I modified the parameter in the file 'NRELOffshrBsline5MW_OC3Hywind_MAP.dat', and then ran the openfast, but I get the following error message:
FAST_InitializeAll:MAP_Init:MAP_FATAL[12] : Failed to convert diameter parameter in the cable 
library to a double. Check the MAP input file. 
MAP_FATAL[16] : Could not complete the initialization process in MAP_Init() beause of either 
syntax errors in the input file or memory allocation problems. 
MAP_ERROR[7] : Failed to write node information to the MAP summary file. 
MAP_FATAL[37] : Failed to write the MAP summary file. 
I am by rewriting the  'NRELOffshrBsline5MW_OC3Hywind_MAP.dat' to achieve parameter modification, my MATLAB code is as follows
function []= MooringRenew(chrom)
if chrom(14)<=10+sqrt((chrom(13)-10-5.2)^2+(320-(0.5*chrom(3)+12))^2)
chrom(14)=10+sqrt((chrom(13)-10-5.2)^2+(320-(0.5*chrom(3)+12))^2)+1;
end
fid = fopen('E:\OCEANDESIGN\NSGAII-openfast\OpenFast\openfast-main\openfast-main\reg_tests\r-test\glue-codes\openfast\5MW_OC3Spar_DLL_WTurb_WavesIrr\NRELOffshrBsline5MW_OC3Hywind_MAP.dat','wt');
fprintf(fid,'%s \n','---------------------- LINE DICTIONARY ---------------------------------------');
fprintf(fid,'%s \n','LineType Diam MassDenInAir EA CB CIntDamp Ca Cdn Cdt');
fprintf(fid,'%s \n','---------------------- LINE DICTIONARY ---------------------------------------');
fprintf(fid,'%s \n','(-) (m) (kg/m) (N) (-) (Pa-s) (-) (-) (-)');
fprintf(fid,'%s\t %03.2f\t %f\t %f\t %f\t %f\t %f\t %f\t %f\t \n','Material',0.09,77.7066,384.243E6,0.001,1.0E8,0.6,-1.0,0.05);
fprintf(fid,'%s \n','---------------------- NODE PROPERTIES ---------------------------------------');
fprintf(fid,'%s \n','Node Type X Y Z M B FX FY FZ');
fprintf(fid,'%s \n','(-) (-) (m) (m) (m) (kg) (mˆ3) (N) (N) (N)');
fprintf(fid,'%d \t %s \t %13.7f \t %13.7f \t %s\t %d \t %d \t %s\t %s \t %s\n',1,'fix',chrom(13)*cosd(0),chrom(13)*sind(0),'depth',0,0,'#','#','#');
fprintf(fid,'%d \t %s \t %13.7f \t %13.7f \t %d \t %d \t %d \t %s \t %s\t %s\n',2,'vessel',5.2,0.0,-0.5*chrom(3)-12,0,0,'#','#','#');
fprintf(fid,'%s \n','---------------------- LINE PROPERTIES ---------------------------------------');
fprintf(fid,'%s \n','Line LineType UnstrLen NodeAnch NodeFair Flags');
fprintf(fid,'%s \n','(-) (-) (m) (-) (-) (-)');
fprintf(fid,'%d \t %s \t %3.2f \t %d \t %d \t %s \t \n',1,'Material',chrom(14),1,2,'tension_fair tension_anch');
fprintf(fid,'%s \n','---------------------- SOLVER OPTIONS-----------------------------------------');
fprintf(fid,'%s \n','Option');
fprintf(fid,'%s \n','(-)');
fprintf(fid,'%s \n','repeat 240 120');
sta=fclose("all");
end

**Comment by jjonkman** [comment]
Dear @sjihf12,

The error points to an error in your MAP++ input file format.  I haven't fully reviewed your MATLAB script, but I do see that you have the "LINE DICTIONARY" line showing up twice. 

Best regards.

**Comment by sjihf12** [comment]
Thank you!
